### ISSUE TITLE
Publish docker-compose to the Docker's repository

### ISSUE THREAD

**Issue by bugeaud** [docker/compose#6135]
Docker has chosen to publish to a dedicated repository its releases inside download.docker.com for the various Linux distributions.

Docker Compose should publish its releases to this repository for consistency purpose of the whole stack.

This would ease the maintenance of Docker Compose related applications and ensure version update & security fixes beeing applied.

OS : Any distrib supported as download.docker.com

**Comment by thackel** [comment]
Any news here?
The proposed way of installation by "curl-ing" the releases into some system directory is somehow an antipattern for distributions with packagemanagement.

**Comment by BenjamenMeyer** [comment]
@thackel  agree - a `curl` based install should *never* be the primary method of install...that's just asking for security violations to occur

**Comment by ndeloof** [comment]
I think a major difference between docker as distributed by download.docker.com and docker-compose released under github is that the later is only a command line tool you can install under `~/bin` without requirement for specific privileges, so that a `curl` based installation is not such a big deal.

Anyway noted on https://github.com/docker/compose/pull/6932/files#r332869532 so this can be considered as we revisit the release process

**Comment by ndeloof** [comment]
duplicates https://github.com/docker/compose/issues/2235

**Comment by bugeaud** [comment]
Nicolas, #2235 is only about building .deb and .rpm it does not fit with #6135 aim.

This issue is more broad : docker setup is done one way and docker compose another way. Both must be in line. I've opened this issue as I foreseen that Docker was the "master" in the way of proceeding setups. I've removed explicit OS reference in the description avoid further quiproquo on this.

Anyone feeling that Docker is doing it the bad way shall open an issue there.

So IMHO is shall be neither closed not marked down as a duplicate.

Rgs,
JB

**Comment by ndeloof** [comment]
Right, my point here is that we want to improve docker-compse delivery, which includes :
- publish artifacts under download.docker.com aside other docker binaries
- sign artifacts
- offer Linux distribution system packages

We can keep this issue open if you wish, big I think better to group related ones together within a more visible epic.

**Comment by cefn** [comment]
> @thackel agree - a `curl` based install should _never_ be the primary method of install...that's just asking for security violations to occur

Having a curl-based install as 'primary', while docker-compose is available in python's pip (a secured software repository) is surprising. To install through pip, (assuming you have [installed pip3](https://vgkits.org/blog/install-pip3-on-linux/)), you just...

sudo pip3 install docker-compose

I think for most Linux users, the documentation describing how to use pip is just inaccessible, making the curl install 'primary' since there are four alternatives provided at https://docs.docker.com/compose/install/ at the time of writing...

> Windows, Mac OS, Linux, Alternative Install Options

So if a Linux user does the normal thing and follows the linux link, they won't find a guide to install via the pip software repository, which is lurking under 'Alternative Install Options'. 

The fact curl is improperly the 'primary' method for Linux is potentially a documentation issue rather than a packaging one. I agree a PPA and .deb might be ideal for e.g. Ubuntu/Debian users but it is already 'packaged' via pip but just not easy to find.
